connectors/common/base.go

```diff
@@ -468,66 +476,90 @@ func (c *connector) StartReadToChannel(flowId iface.FlowID, options iface.Connec
 					ns := iface.Namespace{Db: task.Def.Db, Col: task.Def.Col}
 					c.progressTracker.TaskStartedProgressUpdate(ns, task.Id)
 
-					var cursor []byte
-
-					var pCursor []byte
-					if task.Def.Low != nil {
-						pCursor = task.Def.Low.(primitive.Binary).Data
-					}
-					for {
-						res, err := c.maybeOptimizedImpl.ListData(c.flowCtx, connect.NewRequest(&adiomv1.ListDataRequest{
-							Partition: &adiomv1.Partition{
-								Namespace: sourceNamespace,
-								Cursor:    pCursor,
-							},
-							Type:   c.settings.SourceDataType,
-							Cursor: cursor,
-						}))
-						if err != nil {
-							if !errors.Is(err, context.Canceled) {
-								slog.Error(fmt.Sprintf("Error listing data: %v", err), "task", task.Id)
-							}
-							continue Loop
+					backoffConfig := backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 3)
+					err := backoff.Retry(func() error {
+						if docs > 0 {
+							readerProgress.initialSyncDocs.Add(^uint64(docs - 1))
```

Comment:
I'm a bit confused by this - what does "^uint64(docs - 1)" signify?

Reply:
thats how to do subtraction with the unsigned int